src/GovUk.Education.ExploreEducationStatistics.Content.Api/Services/MethodologyService.cs

```diff
@@ -65,5 +69,105 @@ public class MethodologyService : IMethodologyService
+                .ThenInclude(pm => pm.MethodologyParent)
+                .ThenInclude(m => m.Versions)
+                .ToListAsync();
+
+            return themesWithMethodologies.Where(IsThemePublished)
+                .Select(BuildThemeTree)
+                .OrderBy(theme => theme.Title)
+                .ToList();
+        }
+
+        private static ThemeTree<PublicationMethodologiesTreeNode> BuildThemeTree(Theme theme)
+        {
+            return new ThemeTree<PublicationMethodologiesTreeNode>
+            {
+                Id = theme.Id,
+                Summary = null,
+                Title = theme.Title,
+                Topics = theme.Topics
+                    .Where(IsTopicPublished)
+                    .Select(BuildTopicTree)
+                    .OrderBy(topic => topic.Title)
+                    .ToList()
+            };
+        }
+
+        private static TopicTree<PublicationMethodologiesTreeNode> BuildTopicTree(Topic topic)
+        {
+            return new TopicTree<PublicationMethodologiesTreeNode>
+            {
+                Id = topic.Id,
+                Summary = null,
+                Title = topic.Title,
+                Publications = topic.Publications
+                    .Where(IsPublicationPublished)
+                    .Select(BuildPublicationNode)
+                    .OrderBy(publication => publication.Title)
+                    .ToList()
+            };
+        }
+
+        private static PublicationMethodologiesTreeNode BuildPublicationNode(Publication publication)
+        {
+            return new PublicationMethodologiesTreeNode
+            {
+                Id = publication.Id,
+                Title = publication.Title,
+                Summary = publication.Summary,
+                Slug = publication.Slug,
+                Methodologies = publication.Methodologies.Select(BuildMethodologyForLatestVersion).ToList()
+            };
+        }
+
+        private static bool IsThemePublished(Theme theme)
+        {
+            return theme.Topics.Any(IsTopicPublished);
+        }
+
+        private static bool IsTopicPublished(Topic topic)
+        {
+            return topic.Publications.Any(IsPublicationPublished);
+        }
+
+        private static bool IsPublicationPublished(Publication publication)
+        {
+            // TODO SOW4 Potentially remove this check on Releases in future
+            var hasReleases = publication.Releases.Any(release => release.IsLatestPublishedVersionOfRelease());
+
+            if (hasReleases)
+            {
+                // TODO SOW4 There should be no LatestPublishedVersion of a Methodology returned if the Publication has no published releases
+                // removing the need for the outer check on hasReleases
+                var hasMethodologies = publication.Methodologies.Any(pm =>
+                    pm.MethodologyParent.LatestPublishedVersion() != null);
+                return hasMethodologies;
```

Comment:
Why does having published methodologies indicate that a Publication is published?  Does this mean that if it had lots of published Releases but still the Publication didn't have a Methodology attached, the Publication would be considered not published?

Reply:
Yes that's correct, just in the context of producing this Publication tree of methodologies, we say a Publication isn't published so that its not included in the result if it has no Methodologies.

The `IsXPublished` methods are filtering out nodes which don't have any leaf (methodology) nodes so no blank Themes/Topics/Publications are listed on the Methodologies page.

I believe the front end is also doing this kind of filtering although it shouldn't be required because there should never be a Theme/Topic/Publication with no methodologies returned by this service.

I think this method name made sense when producing the Publications tree for Releases, but not here so I'll rename it to something like `IsPublicationNodeIncluded`